lib/spack/spack/modules/common.py

```diff
@@ -750,6 +756,14 @@ def write(self, overwrite=False):
         with open(self.layout.filename, 'w') as f:
             f.write(text)
 
+        # Set the file permissions of the module to match that of the package
+        # We have to check for file existence here in case the Writer has been
+        # connected to a StringIO
+        if os.path.exists(self.layout.filename):
+            llnl.util.filesystem.chmod_x(self.layout.filename, self.perms)
+            if self.group:
+                llnl.util.filesystem.chgrp(self.layout.filename, self.group)
```

Comment:
Can we consolidate the permission setting logic in one place?  i.e., both this class and the existing package existing permission-setting logic need to know how to set permissions on a file based on a spec.  So if we had a `SpecPermissionSetter` class somewhere, or just a `set_permissions_by_spec(spec, path)` function, we could avoid having to `chmod`/`chgrp` everywhere.